### PR TITLE
Add readme

### PR DESCRIPTION
**Pull request by MonicaisHer**
Signed-off-by: Mengyi Wang<mengyi.wang@canonical.com>

### CODE REVIEW

File: README.md
Comment (farshidtz) [inline README.md]: I'm not sure if this information should be in the source code. Is this hardware configured in the source?

File: README.md
Comment (farshidtz) [inline README.md]: This isn't necessary. It would be invalid in forks and github already has this.

File: README.md
Comment (farshidtz) [inline README.md]: The git clone command isn't necessary. It would be invalid in forks and github already has this.

File: README.md
Comment (farshidtz) [inline README.md]: I think many of the badges are redundant. All of tag, license, pull req, contributors, committers, commit activity are already available on github.

**Review by farshidtz — CHANGES_REQUESTED** [review]

**Comment by farshidtz** [comment]
Thanks a lot, the readme is very good. I added some minor inline comments.

I'm yet to test the commands and provide feedback on those.

File: README.md
Comment (MonicaisHer) [reply README.md]: No. I think they are configured by checkbox team. Would you like me to remove this table?

File: README.md
Comment (MonicaisHer) [inline README.md]: Change this to uppercase

File: README.md
Comment (farshidtz) [reply README.md]: Yes, I think it is better to remove.

**Comment by farshidtz** [comment]
I noticed that the README is missing instructions for running the tests in the snap. This would be very useful to run tests in isolation. It should actually be the preferred way to run tests.

File: README.md
Comment (MonicaisHer) [reply README.md]: Removed. Thanks.

File: README.md
Comment (MonicaisHer) [reply README.md]: Removed.

File: README.md
Comment (MonicaisHer) [reply README.md]: Removed.

File: README.md
Comment (MonicaisHer) [reply README.md]: Removed.

**Comment by MonicaisHer** [comment]
@farshidtz  Thanks a lot for your review. I have add a `Run tests using checkbox-edgexfoundry snap in isolation (recommend)` section.

File: README.md
Comment (farshidtz) [inline README.md:7]: `/../../` this should be an absolute URL to direct people looking at Launchpad mirror to Github.

Comment (farshidtz) [inline README.md]: `/../../` this should be an absolute URL to direct people looking at Launchpad mirror to Github.

I suggest keeping the original inline URLs because Launchpad doesn't support markdown.

File: README.md
Comment (farshidtz) [inline README.md]: How is this relevant? Some setup issue missing in tests?

File: README.md
Comment (farshidtz) [inline README.md]: Please remove this so that this PR doesn't have to wait for the LICENSE addition. It is not necessary to be here.
```suggestion
```

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
This project contains the checkbox tests of the [Edgex Foundry](https://docs.edgexfoundry.org/) snaps.
```

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
When edgexfoundry snap is released to a $TRACK/beta channel, the corresponding checkbox test will be triggered. 
```

File: README.md
Comment (farshidtz) [inline README.md]: aren't all the tests instructions here to run them locally?

Comment (farshidtz) [inline README.md]: Please use short section to improve readability and ToC. Details can go under it.
- Usage
  - Run using checkbox-edgexfoundry snap
  - Run using Checkbox CLI
  - Run test scripts directly

File: README.md
Comment (farshidtz) [inline README.md]: 1. Explanation should be in the body. E.g.: This is the recommended method as it runs all the tests in isolation.
2. Where to get it? -> https://snapcraft.io/checkbox-edgexfoundry
3. How is it built? -> https://launchpad.net/~ce-certification-qa/+snap/checkbox-edgexfoundry-edge
4. How should it be installed? -> Development Mode + example command

File: README.md
Comment (farshidtz) [inline README.md]: Is it possible to place the information in textual form in a code block? It would be better for version control.

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
Enter the desired test directory:
```

File: README.md
Comment (farshidtz) [inline README.md]: The last three lines are really enough, followed by one or two examples:

Get a list of available options:
```
sudo ./run-all-tests-locally.sh -h
```

For example, to run a single test with a local snap:
```bash
sudo ./run-all-tests-locally.sh -s edgexfoundry.snap -t test-rules-engine.sh
```

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
- The current tests plan only covers the [edgexfoundry](https://snapcraft.io/edgexfoundry) snap. It does not cover any of the device or app service snaps.
```

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
```

**Review by farshidtz — CHANGES_REQUESTED** [review]

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
- Mandatory tests: see [units/test-plan.pxu#L113](./units/test-plan.pxu#L113)
```

File: README.md
Comment (farshidtz) [reply README.md]: Scroll down and press SPACE to select the desired test plan:
```
 Select test plan
┌─────────────────────────────────────────────────┐
│    ( ) Dock Hot Plug tests                      │
│    ( ) EdgeX Fuji                               │
│    ( ) EdgeX Geneva                             │
│    ( ) EdgeX Hanoi                              │
│    (X) EdgeX Ireland                            │
│    ( ) EdgeX Jakarta                            │
│    ( ) Firewire tests                           │
└─────────────────────────────────────────────────┘
 Press <Enter> to continue                (H) Help
```

File: README.md
Comment (farshidtz) [inline README.md]:
```suggestion
- Test the installation of edgexfoundry snap
- Test security services proxy certs work properly
- Test that all services can be started properly
- Test that config paths don't include previous snap revision after installation
- Test that config paths don't include previous snap revision after refresh
- Test that services start after refreshing
- Test that services start after refreshing to the same revision
- Test that the system management agent works with the snap
- Test that services are not listening on external network interfaces
- Test that the rules engine works with the snap
```

**Comment by siggiskulason** [comment]
Excellent. Can we then delete the old draft document at https://docs.google.com/document/d/1KK3vfxANapKqkzNlP_ai1yczlvp3gfqQnvhRXv-v12Y/edit#heading=h.qjjnvsuim13r ? Is there anything in there that's not covered here?

**Review by farshidtz — APPROVED** [review]

**Comment by farshidtz** [comment]
Thanks. I'm going to merge and defer minor polishing to another PR.

Thanks a lot @MonicaisHer

**Comment by MonicaisHer** [comment]
@siggiskulason I think this README covered your HOWTO documentation. Thank you!

**Comment by farshidtz** [comment]
I forgot to squash before merging and that created a mess. I force pushed a squashed version.

**Pull request merged**
